src/groups.ts

```diff
@@ -67,12 +72,23 @@ export const getSelectedGroupIds = (appState: AppState): GroupId[] =>
  */
 export const selectGroupsForSelectedElements = (
   appState: AppState,
-  elements: readonly NonDeleted<ExcalidrawElement>[],
+  elements: readonly NonDeletedExcalidrawElement[],
   prevAppState: AppState,
+  /**
+   * supply null in cases where you don't have access to App instance and
+   * you don't care about optimizing selectElements retrieval
+   */
+  app: AppClassProperties | null,
```

Comment:
Also, `elements` are coming from `this.scene.getNonDeletedElements()`, `selectedElements` are coming `appState.scene.getSelectedElements()`, these should be unified. If we don't have the App/Scene instance in all the cases, it might be better to not inject the instance into the function at all and call it before. Potentially it might also make sense splitting it into two functions (old - TODO/to-be-deprecated, new - with injected App/Scene and access to the instance).

Reply:
good point on passing the `Scene` instead of `App`. In this case it makes more sense. Though in others I wanted to later reuse `App` to get `appState`/`elements` like you mention.

But that doesn't apply here — the `elements` aren't necessarily always coming from `this.scene.getNonDeletedElements()`, but could be the _next_ elements to be set to state (but aren't yet), hence why the `app`/`Scene` is also optional as it would not make sense to attempt to retrieve the selection from cache for elements that are not yet stored in `Scene`.